modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/SelectStreamsWorkflowOperationHandler.java

```diff
@@ -0,0 +1,529 @@
+  private static final Logger logger = LoggerFactory.getLogger(SelectStreamsWorkflowOperationHandler.class);
+
+  /** Name of the 'encode to video only work copy' encoding profile */
+  private static final String PREPARE_VIDEO_ONLY_PROFILE = "video-only.work";
+
+  /** Name of the muxing encoding profile */
+  private static final String MUX_AV_PROFILE = "mux-av.work";
+
+  /** The composer service */
+  private ComposerService composerService = null;
+
+  /** The local workspace */
+  private Workspace workspace = null;
+
+  /** The configuration options for this handler */
+  private static final SortedMap<String, String> CONFIG_OPTIONS;
+
+  private enum AudioMuxing {
+    NONE, FORCE, DUPLICATE;
+
+    @Override
+    public String toString() {
+      return super.toString().toLowerCase();
+    }
+
+    static AudioMuxing fromConfigurationString(final String s) {
+      return AudioMuxing.valueOf(s.toUpperCase());
+    }
+  }
+
+  private static final String CONFIG_AUDIO_MUXING = "audio-muxing";
+
+  private static final String CONFIG_FORCE_TARGET = "force-target";
+
+  private static final String FORCE_TARGET_DEFAULT = "presenter";
+
+  static {
+    CONFIG_OPTIONS = new TreeMap<>();
+    CONFIG_OPTIONS.put("source-flavor", "The \"flavor\" of the track to use as a video source input");
+    CONFIG_OPTIONS.put("target-flavor", "The flavor to apply to the encoded file");
+    CONFIG_OPTIONS.put("target-tags", "The tags to apply to the encoded file");
+    CONFIG_OPTIONS.put(CONFIG_FORCE_TARGET,
+            String.format("Target flavor type for the \"%s\" option \"%s\" (default %s)", CONFIG_AUDIO_MUXING,
+                    AudioMuxing.FORCE, FORCE_TARGET_DEFAULT));
+    CONFIG_OPTIONS.put(CONFIG_AUDIO_MUXING,
+            String.format("Either \"%s\", \"%s\" or \"%s\" to specially mux audio streams", AudioMuxing.NONE,
+                    AudioMuxing.DUPLICATE, AudioMuxing.FORCE));
+  }
+
+  @Override
+  public SortedMap<String, String> getConfigurationOptions() {
+    return CONFIG_OPTIONS;
+  }
+
+  /**
+   * Callback for the OSGi declarative services configuration.
+   *
+   * @param composerService
+   *          the local composer service
+   */
+  protected void setComposerService(final ComposerService composerService) {
+    this.composerService = composerService;
+  }
+
+  /**
+   * Callback for declarative services configuration that will introduce us to the local workspace service.
+   * Implementation assumes that the reference is configured as being static.
+   *
+   * @param workspace
+   *          an instance of the workspace
+   */
+  public void setWorkspace(final Workspace workspace) {
+    this.workspace = workspace;
+  }
+
+  private EncodingProfile getProfile(final String identifier) {
+    final EncodingProfile profile = this.composerService.getProfile(identifier);
+    if (profile == null) {
+      throw new IllegalStateException(String.format("couldn't find encoding profile \"%s\"", identifier));
```

Comment:
IIRC there's an exception subclass floating around somewhere for this.

Reply:
Ah yes, the other WOH use `WorkflowOperationException`, so I'll adapt this.